atomisticparsers/gromacs/parser.py

```diff
@@ -180,72 +181,101 @@ def str_to_step_info(val_in):
-            sampling_method=sampling_method,
-            integrator_type=integrator,
-            ensemble_type=ensemble_type,
-        )
-
-    def get_tpstat_settings(self):
-        input_parameters = self.get("input_parameters", {})
-        target_t = input_parameters.get("ref-t", 0) * ureg.kelvin
 
-        thermostat_type = None
-        tcoupl = input_parameters.get("tcoupl", "no").lower()
-        if tcoupl != "no":
-            thermostat_type = (
-                "Velocity Rescaling" if tcoupl == "v-rescale" else tcoupl.title()
-            )
+class GromacsMdpParser(TextParser):
+    def __init__(self):
+        super().__init__(None)
 
-        thermostat_tau = input_parameters.get("tau-t", 0) * ureg.ps
+    def init_quantities(self):
+        def str_to_input_parameters(val_in):
+            re_array = re.compile(r"\s*([\w\-]+)\[[\d ]+\]\s*=\s*\{*(.+)")
+            re_scalar = re.compile(r"\s*([\w\-]+)\s*[=:]\s*(.+)")
+            parameters = dict()
+            val = val_in.split("\n")
+            val = [line.strip() for line in val]
+            for val_n in val:
+                val_scalar = re_scalar.match(val_n)
+                if val_scalar:
+                    parameters[val_scalar.group(1)] = val_scalar.group(2)
+                    continue
+                val_array = re_array.match(val_n)
+                if val_array:
+                    parameters.setdefault(val_array.group(1), [])
+                    value = [
+                        float(v) for v in val_array.group(2).rstrip("}").split(",")
+                    ]
+                    parameters[val_array.group(1)].append(
+                        value[0] if len(value) == 1 else value
+                    )
+            return parameters
 
-        # TODO infer langevin_gamma [s] from bd_fric
-        # bd_fric = self.get('bd-fric', 0, unit='amu/ps')
-        langevin_gamma = None
+        self._quantities = [
+            Quantity(
+                "input_parameters",
+                r"([\s\S]+)",
+                str_operation=str_to_input_parameters,
+            ),
+        ]
 
-        target_p = input_parameters.get("ref-p", 0) * ureg.bar
-        # if P is array e.g. for non-isotropic pressures, get average since metainfo is float
-        if hasattr(target_p, "shape"):
-            target_p = np.average(target_p)
 
-        barostat_type = None
-        pcoupl = input_parameters.get("pcoupl", "no").lower()
-        if pcoupl != "no":
-            barostat_type = pcoupl.title()
+class GromacsXvgParser(TextParser):
+    def __init__(self):
+        super().__init__(None)
 
-        barostat_tau = input_parameters.get("tau-p", 0) * ureg.ps
+        def str_to_results(val_in):
+            results = {
+                "column_vals": None,
+                "title": "",
+                "xaxis": "",
+                "yaxis": "",
+                "column_headers": [],
+            }
+            re_columns = re.compile(r"@\s*s\d{1,2}\s*legend\s*\".*\"")
```

Comment:
i would put the regex patterns in init so you do not compile everytime

Reply:
done.